src/util/psbt/map/mod.rs

```diff
@@ -20,18 +20,29 @@ use consensus::encode;
 use util::psbt;
 use util::psbt::raw;
 
-/// A trait that describes a PSBT key-value map.
-pub trait Map {
+mod global;
+mod input;
+mod output;
+
+pub(super) use self::input::Input;
+pub(super) use self::global::consensus_decode_global;
+pub use self::output::{Output, TapTree};
+
+/// A trait for inserting into a PSBT key-value map.
+pub(super) trait Insert {
```

Comment:
Not strong feelings about it, but having two names so different does not seem idiomatic (in a sense the e.g. standard library would do that).

https://doc.rust-lang.org/std/ops/trait.DerefMut.html is a best example, but not exactly like what is happening here.

At very least it seems `MapInsert`+ `Map` seems to tie the two together.

Also, unless there's a good use-case for maps one can insert to but not get, `trait MapInsert : Map` would help the code using these, as `T : MapInsert` will imply `T: Map` which will save some typing when both reading and writing is used.

Reply:
Thanks for your input @dpc, I ended up going with @apoelstra's suggestion below and making `insert_pair` into functions.